src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx

```diff
@@ -57,19 +57,19 @@ const VolcanoPlot = ({ experimentId }) => {
 
   useEffect(() => {
     dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
-  }, []);
-
-  useEffect(() => {
-    // Sync plot and settings with last used config
     setComparisonGroup({
       ...plotCellSets,
       type: plotComparisonType,
     });
+  }, []);
 
-    // Show plot using last shown data
-    if (data.length === 0) return;
+  useEffect(() => {
+    if (!config || !data.length) return;
     setDataPointStatus();
```

Comment:
Can you refactor `setDataPointStatus()` into something like `setPlotData(calculateDataPointStatus(data))`? That way it's much clearer what the function does, unlike how it's written right now.